### PR TITLE
[scaling] feat: builder service

### PR DESCRIPTION
**Pull request by chesedo**
## Description of change
A builder service for our new scaling project

## How has this been tested? (if applicable)
With new tests that were added in the code

### CODE REVIEW

File: builder/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+}
+
+/// Make a `flake.nix` file at the given path
+fn build_flake_file(path: &Path) -> Result<(), Error> {
+    let mut package = cargo::Package::from_current_dir(path)?;
+    package.resolve();
+
+    let package: nix::Package = package.into();
+    // let name = package.name;
+    let name = "test";
```
Comment (oddgrd) [inline builder/src/lib.rs:161]: I think you forgot this one :smile:

File: builder/tests/integration_tests.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        let mut client = BuilderClient::connect(format!("http://localhost:{port}"))
+            .await
+            .unwrap();
+        let resources = Path::new(env!("CARGO_MANIFEST_DIR"))
+            .join("tests")
+            .join("resources");
+
+        // Build a normal hello world archive
+        let deployment_id = Ulid::new().to_string();
+        let archive = fs::read(resources.join("hello-world-0.1.0.crate")).unwrap();
```
Comment (oddgrd) [inline builder/tests/integration_tests.rs:44]: Oh, does the new builder take .crate files? Does that mean we need to follow the cargo package restrictions again?

**Review by oddgrd — APPROVED** [review]
Having this standardized design for services makes reviewing a lot easier! :partying_face: LGTM! Just one question about the builder.

File: builder/tests/integration_tests.rs (hunk 2 above)
Comment (chesedo) [reply builder/tests/integration_tests.rs:44]: Oh, no. A `.crate` is just a `tar.gz`. I'll update the test :sweat_smile:

**Pull request merged**